File: Java/Number-of-Good-Pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        int numGoodPairs = 0;
        
        for (int i = 0; i < nums.length; i++) {
            for (int j = 0; j < nums.length; j++) {
                if (i == j) continue; // cannot check against itself
                if (nums[i] == nums[j] && i < j) numGoodPairs++;
            }
        }
        
        return numGoodPairs;
    }
}

File: C++/Number of Good Pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        unordered_map<int,int> mp;
        for(auto &i : nums) {
            mp[i]++;
        }
        int ans = 0;
        for(auto & i : mp) {
            ans += (i.second) * (i.second - 1) >> 1;
        }
        return ans;
    }
};
